### DISCUSSION TITLE
superRefine Error Path

### DISCUSSION THREAD

**Discussion by carlBgood** [colinhacks/zod#2035] [Q&A]
If I have a schema like the following...

`z.object({
  name : z.string(),
  description: z.string(),
  permissions: z.union([z.string(), z.literal('')]
})`

I'm running a custom superRefine and trying to add an error to a specific field in the permissions object.

The object key (the field name) is held in `pID`. But when trying to set an error on that field, it just creates an array on the 'permissions' path. This is the ctx:

`ctx.addIssue({
  code: z.ZodIssueCode.custom,
  message: `not cool bro`,
  path: ['password', pID],
  fatal: true,
}) `

The error _always_ comes up on `fieldErrors > permissions > 0` instead of 'fieldErrors > permissions.{pID string}`

What am I missing here?

**Comment by JacobWeisenburger** [comment]
> `z.object({ name : z.string(), description: z.string(), permissions: z.union([z.string(), z.literal('')] })`
> 
> I'm running a custom superRefine and trying to add an error to a specific field in the permissions object.

`permissions` seems to be `string | ''`. So I'm not sure how you could be treating it as an object.

Please send a full reproducible example.
